test/Bound.lean

```diff
@@ -0,0 +1,140 @@
+lemma test_nat : (n : ℝ) ≥ 0 := by bound
+lemma test_num : 0 ≤ 7 := by bound
+lemma test_num_real : 0 ≤ (7 : ℝ) := by bound
+lemma test_zero_one : 0 ≤ (1 : ℝ) := by bound
+lemma test_nnreal : 0 ≤ (u : ℝ) := by bound
+lemma test_zero : 0 ≤ (0 : ℝ) := by bound
+lemma test_pow : 0 ≤ 2^n := by bound
+lemma test_inv : 0 ≤ (0 : ℝ)⁻¹ := by bound
+end nonneg_tests
+
+section bound_tests
+variable {a b c x y : ℝ}
+variable {z : ℂ}
+variable {n : ℕ}
+lemma test_sq (n : x ≥ 0) (h : x ≤ y) : x^2 ≤ y^2 := by bound
+lemma test_sq_ge (n : x ≥ 0) (h : x ≤ y) : y^2 ≥ x^2 := by bound
+lemma test_mul_left (n : a ≥ 0) (h : x ≤ y) : a * x ≤ a * y := by bound
+lemma test_mul_right (n : a ≥ 0) (h : x ≤ y) : x * a ≤ y * a := by bound
+lemma test_mul_both (bp : b ≥ 0) (xp : x ≥ 0) (ab : a ≤ b) (xy : x ≤ y) : a * x ≤ b * y := by bound
+lemma test_abs_mul (h : x ≤ y) : abs z * x ≤ abs z * y := by bound
+lemma test_add_left (h : x ≤ y) : a + x ≤ a + y := by bound
+lemma test_add_right (h : x ≤ y) : x + a ≤ y + a := by bound
+lemma test_add_both (ab : a ≤ b) (xy : x ≤ y) : a + x ≤ b + y := by bound
+lemma test_sub_left (h : x ≥ y) : a - x ≤ a - y := by bound
+lemma test_sub_right (h : x ≤ y) : x - a ≤ y - a := by bound
+lemma test_sub_both (ab : a ≤ b) (xy : x ≥ y) : a - x ≤ b - y := by bound
+lemma test_sub_pos (h : x < y) : y - x > 0 := by bound
+lemma test_le_of_lt (h : x > 0) : x ≥ 0 := by bound
+lemma test_extra (f : ℕ → ℝ) (h : ∀ n, f n ≥ 0) : f n ≥ 0 := by bound [h n]
+lemma test_1_4 : (1 : ℝ) < 4 := by bound
+lemma test_2_4 : (2 : ℝ) < 4 := by bound
+lemma test_div_left (hc : c ≥ 0) (h : a ≤ b) : a / c ≤ b / c := by bound
+lemma test_div_right (ha : a ≥ 0) (hc : c > 0) (h : b ≥ c) : a / b ≤ a / c := by bound
+lemma test_coe (x y : ℝ≥0) (h : x < y) : (x : ℝ) < y := by bound
+lemma test_dist : dist a c ≤ dist a b + dist b c := by bound
+lemma test_log (x y : ℝ) (x0 : 0 < x) (h : x ≤ y) : x.log ≤ y.log := by bound
+end bound_tests
+
+section guess_tests
+variable {a b c : ℝ}
+variable {n m : ℕ}
+lemma test_le_max_of_le_left (h : a ≤ b) : a ≤ max b c := by bound
+lemma test_le_max_of_le_right (h : a ≤ c) : a ≤ max b c := by bound
+lemma test_lt_max_of_lt_left (h : a < b) : a < max b c := by bound
+lemma test_lt_max_of_lt_right (h : a < c) : a < max b c := by bound
+lemma test_min_le_of_left_le (h : a ≤ c) : min a b ≤ c := by bound
+lemma test_min_le_of_right_le (h : b ≤ c) : min a b ≤ c := by bound
+lemma test_min_lt_of_left_lt (h : a < c) : min a b < c := by bound
+lemma test_min_lt_of_right_lt (h : b < c) : min a b < c := by bound
+lemma test_pow_le_pow_right (a1 : 1 ≤ a) (h : m ≤ n) : a^m ≤ a^n := by bound
+lemma test_pow_le_pow_of_le_one (a0 : 0 ≤ a) (a1 : a ≤ 1) (h : n ≤ m) : a^m ≤ a^n := by bound
+lemma test_rpow_le_rpow_of_exponent_le (a1 : 1 ≤ a) (h : b ≤ c) : a^b ≤ a^c := by bound
+lemma test_rpow_le_rpow_of_exponent_ge (a0 : 0 < a) (a1 : a ≤ 1) (h : c ≤ b) : a^b ≤ a^c := by bound
+end guess_tests
+
+-- This breaks without appropriate g.withContext use
```

Comment:
```suggestion
-- This breaks without appropriate `g.withContext` use in <function-name>
```

Reply:
The broken code was before the `aesop` port. Okay if I leave it in there with this comment?

```
/-- This broke without appropriate `g.withContext` use in an older implementation of `bound`.
Leaving the test here just in case. -/
```